apollo-openapi/src/main/java/com/ctrip/framework/apollo/openapi/client/service/ItemOpenApiService.java

```diff
@@ -189,4 +193,41 @@ public void removeItem(String appId, String env, String clusterName, String name
     }
 
   }
+
+  @Override
+  public OpenPageDTO<OpenItemDTO> findItemsByNamespace(String appId, String env, String clusterName,
+                                                       String namespaceName, int page, int size) {
+    if (Strings.isNullOrEmpty(clusterName)) {
+      clusterName = ConfigConsts.CLUSTER_NAME_DEFAULT;
+    }
+    if (Strings.isNullOrEmpty(namespaceName)) {
+      namespaceName = ConfigConsts.NAMESPACE_APPLICATION;
+    }
+
+    checkNotEmpty(appId, "App id");
+    checkNotEmpty(env, "Env");
+    checkPage(page);
+    checkSize(size);
+
+    OpenApiPathBuilder pathBuilder = OpenApiPathBuilder.newBuilder()
+            .envsPathVal(env)
+            .appsPathVal(appId)
+            .clustersPathVal(clusterName)
+            .namespacesPathVal(namespaceName)
+            .itemsPathVal("")
+            .addParam("page", page)
+            .addParam("size", size);
+
+    try (CloseableHttpResponse response = get(pathBuilder)) {
+      Type type = new TypeToken<OpenPageDTO<OpenItemDTO>>(){}.getType();
+      return gson.fromJson(EntityUtils.toString(response.getEntity()), type);
+    } catch (Throwable ex) {
+      // return null if item doesn't exist
+      if (ex instanceof ApolloOpenApiException && ((ApolloOpenApiException)ex).getStatus() == 404) {
+        return null;
+      }
```

Comment:
It's better to throw the exception instead of returning a null object for OpenPageDTO.
```suggestion
```

Reply:
Yes, return null is indeed prone to problem.